src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AddHandler.scala

```diff
@@ -53,4 +53,69 @@ class AddHandler(authorizationProvider: AuthorizationProvider) extends Actor {
     val method = agoraBusiness.insert(agoraEntity.copy(createDate = Option(new DateTime())), username)
     context.parent ! RequestComplete(spray.http.StatusCodes.Created.intValue, method)
   }
+
+}
+
+case class MethodImportResolver(username: String, business: AgoraBusiness, authorizationProvider: AuthorizationProvider) {
+
+  def importResolver(importUri: String): WdlSource = {
+    ImportResolverHelper.validateUri(importUri)
+    val method = ImportResolverHelper.resolve(importUri, business, username)
+    val authorizedMethod = authorizationProvider.filterByReadPermissions(method, username)
```

Comment:
Thanks for waiting dave. With the new GcsAuthorizationProvider, I think this code should look like

```scala
import org.broadinstitute.dsde.agora.server.dataaccess.acls.gcs.GcsAuthorizationProvider._
if (isAuthorizedForRead(method, username)) method.payload.get
else throw new SyntaxError(s"Can't resolve import: $importUri")
```

Now that I write this, I think the variable name "entity" might be better than "method"